src/meson.build

```diff
@@ -0,0 +1,341 @@
+####
+# Source lists
+####
+source = files(
+   'array.c',
+   'background.c',
+   'board.c',
+   'camera.c',
+   'claim.c',
+   'collision.c',
+   'colour.c',
+   'comm.c',
+   'commodity.c',
+   'cond.c',
+   'conf.c',
+   'console.c',
+   'damagetype.c',
+   'debris.c',
+   'dev.c',
+   'dev_mapedit.c',
+   'dev_outfit.c',
+   'dev_planet.c',
+   'dev_ship.c',
+   'dev_sysedit.c',
+   'dev_system.c',
+   'dev_uniedit.c',
+   'dialogue.c',
+   'economy.c',
+   'env.c',
+   'equipment.c',
+   'escort.c',
+   'event.c',
+   'explosion.c',
+   'faction.c',
+   'fleet.c',
+   'font.c',
+   'glad.c',
+   'gui.c',
+   'gui_omsg.c',
+   'gui_osd.c',
+   'hook.c',
+   'info.c',
+   'input.c',
+   'intro.c',
+   'joystick.c',
+   'land.c',
+   'land_outfits.c',
+   'land_shipyard.c',
+   'land_trade.c',
+   'load.c',
+   'log.c',
+   'map.c',
+   'map_find.c',
+   'map_overlay.c',
+   'map_system.c',
+   'md5.c',
+   'menu.c',
+   'mission.c',
+   'music.c',
+   'music_openal.c',
+   'music_sdlmix.c',
+   'naev.c',
+   'ndata.c',
+   'nebula.c',
+   'news.c',
+   'nfile.c',
+   'nlua.c',
+   'nmath.c',
+   'nopenal.c',
+   'npc.c',
+   'npng.c',
+   'nstring.c',
+   'ntime.c',
+   'nxml.c',
+   'nxml_lua.c',
+   'opengl.c',
+   'opengl_matrix.c',
+   'opengl_render.c',
+   'opengl_shader.c',
+   'opengl_tex.c',
+   'opengl_vbo.c',
+   'options.c',
+   'outfit.c',
+   'pause.c',
+   'perlin.c',
+   'physics.c',
+   'pilot.c',
+   'pilot_cargo.c',
+   'pilot_ew.c',
+   'pilot_heat.c',
+   'pilot_hook.c',
+   'pilot_outfit.c',
+   'pilot_weapon.c',
+   'plasmaf.c',
+   'player.c',
+   'player_autonav.c',
+   'player_gui.c',
+   'queue.c',
+   'rng.c',
+   'save.c',
+   'ship.c',
+   'shiplog.c',
+   'shipstats.c',
+   'slots.c',
+   'sound.c',
+   'sound_openal.c',
+   'sound_sdlmix.c',
+   'space.c',
+   'spfx.c',
+   'start.c',
+   'tech.c',
+   'threadpool.c',
+   'toolkit.c',
+   'unidiff.c',
+   'utf8.c',
+   'weapon.c',
+   'tk/widget/input.c',
+   'tk/widget/image.c',
+   'tk/widget/cust.c',
+   'tk/widget/rect.c',
+   'tk/widget/tabwin.c',
+   'tk/widget/fader.c',
+   'tk/widget/checkbox.c',
+   'tk/widget/imagearray.c',
+   'tk/widget/button.c',
+   'tk/widget/text.c',
+   'tk/widget/list.c',
+)
+
+lua_source = files(
+   'ai.c',
+   'nlua_bkg.c',
+   'nlua_camera.c',
+   'nlua_cli.c',
+   'nlua_col.c',
+   'nlua_commodity.c',
+   'nlua_diff.c',
+   'nlua_evt.c',
+   'nlua_faction.c',
+   'nlua_gfx.c',
+   'nlua_gui.c',
+   'nlua_hook.c',
+   'nlua_jump.c',
+   'nlua_misn.c',
+   'nlua_music.c',
+   'nlua_naev.c',
+   'nlua_news.c',
+   'nlua_outfit.c',
+   'nlua_pilot.c',
+   'nlua_planet.c',
+   'nlua_player.c',
+   'nlua_rnd.c',
+   'nlua_ship.c',
+   'nlua_shiplog.c',
+   'nlua_spfx.c',
+   'nlua_system.c',
+   'nlua_tex.c',
+   'nlua_time.c',
+   'nlua_tk.c',
+   'nlua_var.c',
+   'nlua_vec2.c'
+)
+
+mac_source = files('glue_macos.m')
+
+naev_source = [
+   source,
+   lua_source
+]
+
+# Headers aren't needed for running the builds,
```

Comment:
However, meson does trigger rebuilds on header changes right?

Reply:
Yes, but internally it builds that list by parsing #include statements.